src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,23 +1,34 @@
-import React from "react";
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
+import { getTokenOwner } from "./../../../services/token";
+import { useEscrowContractUsers } from "../../../hooks";
+import { WrappedDocument } from "@govtechsg/open-attestation";
 
-export const TitleTransferPanelContent = () => {
-  const { holderAddress, beneficiaryAddress } = useSelector((state: any) => ({
-    holderAddress: state.token.holderAddress,
-    beneficiaryAddress: state.token.beneficiaryAddress
-  }));
+interface TitleTransferPanelContentProps {
+  document: WrappedDocument;
+}
+
+export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
+  const [tokenOwnerAddress, setTokenOwnerAddress] = useState("");
+  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwnerAddress });
+  const fetchTokenOwner = async () => {
+    try {
+      const address = await getTokenOwner({ document });
+      setTokenOwnerAddress(address);
+    } finally {
+      // prevent code smell in sonar
+    }
+  };
+  fetchTokenOwner();
```

Comment:
do you want to put this into useEffect(()=> {somefunction}, [document]) so that it runs only once everytime document change instead of running once per rerender?